test/e2e/elasticache/smoke_cache_subnet_group.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+
+THIS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+ROOT_DIR="$THIS_DIR/../../.."
+SCRIPTS_DIR="$ROOT_DIR/scripts"
+
+source "$SCRIPTS_DIR/lib/common.sh"
+source "$SCRIPTS_DIR/lib/k8s.sh"
+source "$SCRIPTS_DIR/lib/testutil.sh"
+
+check_is_installed jq
+
+test_name="$( filenoext "${BASH_SOURCE[0]}" )"
+service_name="elasticache"
+ack_ctrl_pod_id=$( controller_pod_id "$service_name")
+debug_msg "executing test: $service_name/$test_name"
+
+aws_resource_name="ack-test-smoke-${service_name}-subnet-gp"
+k8s_resource_name="cachesubnetgroups/$aws_resource_name"
+
+# pre-req: a subnet id to create cache subnet group
+aws_subnet_id="$(aws ec2 describe-subnets --output json | jq -e '.Subnets[0] | .SubnetId')"
+
+if [ $? -eq 4 ] || [ -z $aws_subnet_id ]; then
+    echo "no subnet id found to run the test. Ensure that subnet id is available to run the test."
+    exit 1
+else
+    echo "using subnet id: ${aws_subnet_id} for test."
+fi
+
+
+describe_subnet_json() {
+    jq_expr='.CacheSubnetGroups[] | select(.CacheSubnetGroupName | contains($CacheSubnetGroupName))'
+    aws elasticache describe-cache-subnet-groups --output json | jq -e --arg CacheSubnetGroupName "$aws_resource_name" "$jq_expr"
```

Comment:
Instead of using the jq query here, since the `DescribeCacheSubnetGroups` API call has a filter for the cache subnet group name, you could more easily and efficiently use this:

```
aws elasticache describe-cache-subnet-groups --output json  --cache-subnet-group-name "$aws_resource_name"
```

which will do the filtering on the server side and matches the technique we use in the ECR smoke test here:

https://github.com/aws/aws-controllers-k8s/blob/4a4aae282ba8ebecf25fc8214bd4d58e82de4907/test/e2e/ecr/smoke.sh#L29-L30

Reply:
Thanks for your input.
I have updated the changes.